From 20ef27f0822d28fb399acf532dc2c073db09781b Mon Sep 17 00:00:00 2001
From: Nirmal Raj S <nirm@qti.qualcomm.com>
Date: Thu, 11 Sep 2025 09:37:46 +0530
Subject: [PATCH] handling agent reconnection during gateway reboot

[NOTE] Removed "REMOVE_NETWORK" and "SET_NETWORK" in backhaul_manager

Signed-off-by: Nirmal Raj S <nirm@qti.qualcomm.com>
---
 .../slave/backhaul_manager/backhaul_manager.cpp      | 12 ++++++++----
 .../slave/tasks/ap_autoconfiguration_task.cpp        |  4 ++++
 2 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index 78320bca1..bc18749b0 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -1395,7 +1395,10 @@ bool BackhaulManager::backhaul_fsm_wireless(bool &skip_select)
                 if (!radios_info->sta_wlan_hal || radios_info->sta_iface.empty())
                     continue;
                 std::string iface = radios_info->sta_iface;
+                // PRPLMESH [NLHAL] TODO: Commented down remove network
+#ifndef USE_PRPLMESH_NLHAL
                 radios_info->sta_wlan_hal->disconnect();
+#endif
                 radios_info->sta_wlan_hal->set_4addr_mode(true);
             }
         } else {
@@ -1457,7 +1460,8 @@ bool BackhaulManager::backhaul_fsm_wireless(bool &skip_select)
             db->backhaul.selected_iface_name = iface;
             active_hal                       = get_wireless_hal();
         }
-
+        // PRPLMESH [NLHAL] TODO: Since db is not updated with correct values, commented down set network
+#ifndef USE_PRPLMESH_NLHAL
         if (active_hal->connect(db->device_conf.back_radio.ssid, db->device_conf.back_radio.pass,
                                 db->device_conf.back_radio.security_type,
                                 db->device_conf.back_radio.mem_only_psk, selected_bssid,
@@ -1471,7 +1475,7 @@ bool BackhaulManager::backhaul_fsm_wireless(bool &skip_select)
             FSM_MOVE_STATE(INITIATE_SCAN);
             break;
         }
-
+#endif
         FSM_MOVE_STATE(WIRELESS_ASSOCIATE_4ADDR_WAIT);
         state_attempts           = 0;
         skip_select              = true;
@@ -2067,7 +2071,7 @@ bool BackhaulManager::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t even
         }
 
         // TODO: Need to unite WAIT_WPS and WIRELESS_ASSOCIATE_4ADDR_WAIT handling
-        if (FSM_IS_IN_STATE(WAIT_WPS) || FSM_IS_IN_STATE(WIRELESS_ASSOCIATE_4ADDR_WAIT)) {
+        if (FSM_IS_IN_STATE(WAIT_WPS) || FSM_IS_IN_STATE(WIRELESS_ASSOCIATE_4ADDR_WAIT) || FSM_IS_IN_STATE(WIRELESS_CONFIG_4ADDR_MODE)) {
             auto msg = static_cast<bwl::sACTION_BACKHAUL_CONNECTED_NOTIFICATION *>(data);
             if (!msg) {
                 LOG(ERROR) << "ACTION_BACKHAUL_CONNECTED_NOTIFICATION not found on Connected event";
@@ -2086,7 +2090,7 @@ bool BackhaulManager::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t even
             send_cmdu(m_agent_fd, cmdu_tx);
         }
 
-        if (FSM_IS_IN_STATE(WAIT_WPS)) {
+        if (FSM_IS_IN_STATE(WAIT_WPS) || FSM_IS_IN_STATE(WIRELESS_CONFIG_4ADDR_MODE)) {
             db->backhaul.selected_iface_name = iface;
             db->backhaul.connection_type     = AgentDB::sBackhaul::eConnectionType::Wireless;
             LOG(DEBUG) << "WPS scan completed successfully on iface = " << iface
diff --git a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
index a749250d1..590f36612 100755
--- a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
+++ b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
@@ -266,6 +266,10 @@ void ApAutoConfigurationTask::handle_event(uint8_t event_enum_value, const void
                     continue;
                 }
             }
+     
+            // clear controller discover completed flag and msg_sent flag to send search message 
+            m_discovery_status[radio->wifi_channel.get_freq_type()].completed = false;
+            m_discovery_status[radio->wifi_channel.get_freq_type()].msg_sent = false;
 
             LOG(DEBUG) << "starting discovery sequence on radio_iface=" << radio->front.iface_name;
             FSM_MOVE_STATE(radio->front.iface_name, eState::CONTROLLER_DISCOVERY);
-- 
2.34.1

